feat: Rework details page and make it suitable for mobile devices

lucas-goldner opened the pull request:
Closes https://github.com/alphahorizonio/consumat.io/issues/60

lucas-goldner committed:
feat: Add defaultLayout prop for all pages

lucas-goldner committed:
feat: Add backdrop with desired black filter

lucas-goldner committed:
feat: Add title and description

lucas-goldner committed:
feat: Add relase info card

lucas-goldner committed:
feat: Add cast, directors and genres to details

lucas-goldner committed:
fix: Max Width for different descriptions

lucas-goldner committed:
chore: Delete old version of details page

lucas-goldner committed:
feat: Add title to page title

helmrich reviewed (changes requested):
We already talked about the changes we should make before merging in the meeting. So if you remove the `defaultLayout` prop from the `Layout` component and make minor changes like the addition of a loading spinner to the movie details view we could merge and then continue tweaking/rewriting the detail view and make additions like the watch status, rating functionality, etc. :smile:

helmrich commented on components/Layout.tsx:
```diff
@@ -4,15 +4,22 @@ import Navbar from "./navigation/Navbar";
 };
 
-const Layout = ({ children }: LayoutProps) => {
+const Layout = ({ children, defaultLayout }: LayoutProps) => {
```
We should maybe remove this for now as it's not worth having a property that just affects one page in a wrapper component that's related to all pages in my opinion.

lucas-goldner committed:
Merge branch 'alphahorizonio:main' into main

lucas-goldner committed:
feat: Remove defaultLayout

lucas-goldner commented:
Hey I made the requested changes:
- Added the spinner to the movie details 
- Removed the defaultLayout and changed the styles so it fits better with the new maxWidth now

helmrich reviewed (approved):
LGTM
